src/main/java/school/hei/haapi/model/validator/UserValidator.java

```diff
@@ -0,0 +1,36 @@
+package school.hei.haapi.model.validator;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.validation.ConstraintViolation;
+import javax.validation.Validator;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Component;
+import school.hei.haapi.model.User;
+import school.hei.haapi.model.exception.BadRequestException;
+
+@Component
+@AllArgsConstructor
+public class UserValidator {
+  private final Validator validator;
+
+  private void validateSomeUser(User user) {
+    List<ConstraintViolation<User>> violations = new ArrayList<>(validator.validate(user));
+    if (!violations.isEmpty()) {
+      StringBuilder sb = new StringBuilder();
+      for (int i = 0; i < violations.size(); i++) {
+        sb.append(violations.get(i).getMessage());
+        if (i != violations.size() - 1) {
+          sb.append(" ; ");
+        }
+      }
+      throw new BadRequestException(sb.toString());
+    }
+  }
+
+  public void validateSomeUserList(List<User> users) {
```

Comment:
Actually you should use the same method name. Java is able to find the right method depending on the argument types. This is called overloading, and is possible because Java is strongly typed (you would append `List` in a JS world, but not in Java)